packages/rs-drive-abci/src/execution/validation/state_transition/state_transitions/identity_create/mod.rs

```diff
@@ -81,12 +100,73 @@ impl StateTransitionBasicStructureValidationV0 for IdentityCreateTransition {
     }
 }
 
-impl StateTransitionStateValidationV0 for IdentityCreateTransition {
-    fn validate_state<C: CoreRPCLike>(
+/// A trait for advanced structure validation after transforming into an action
+pub trait StateTransitionStructureKnownInStateValidationForIdentityCreateTransitionV0 {
+    /// Validation of the advanced structure
+    fn validate_advanced_structure_from_state_for_identity_create_transition(
+        &self,
+        action: &IdentityCreateTransitionAction,
+        signable_bytes: Vec<u8>,
+        execution_context: &mut StateTransitionExecutionContext,
+        platform_version: &PlatformVersion,
+    ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error>;
+}
+
+impl StateTransitionStructureKnownInStateValidationForIdentityCreateTransitionV0
+    for IdentityCreateTransition
+{
+    fn validate_advanced_structure_from_state_for_identity_create_transition(
+        &self,
+        action: &IdentityCreateTransitionAction,
+        signable_bytes: Vec<u8>,
+        execution_context: &mut StateTransitionExecutionContext,
+        platform_version: &PlatformVersion,
+    ) -> Result<ConsensusValidationResult<StateTransitionAction>, Error> {
+        match platform_version
+            .drive_abci
+            .validation_and_processing
+            .state_transitions
+            .identity_create_state_transition
+            .advanced_structure
+        {
+            Some(0) => self.validate_advanced_structure_from_state_v0(
+                action,
+                signable_bytes,
+                execution_context,
+                platform_version,
+            ),
+            Some(version) => Err(Error::Execution(ExecutionError::UnknownVersionMismatch {
+                method: "identity create transition: validate_advanced_structure_from_state"
+                    .to_string(),
+                known_versions: vec![0],
+                received: version,
+            })),
+            None => Err(Error::Execution(ExecutionError::VersionNotActive {
```

Comment:
That's because we have Identity TopUp for example which doesn't implement this function so it doesn't have a version. That's why we need to handle None for all methods.